src/colorizer/ColorizeCanvas.ts

```diff
@@ -553,18 +555,15 @@ export default class ColorizeCanvas implements IRenderCanvas {
     this.render();
   }
 
-  /**
-   * Sets the current frame of the canvas, loading the new frame data if the
-   * frame number changes.
-   * @param index Index of the new frame.
-   * @param forceUpdate Force a reload of the frame data, even if the frame
-   * is already loaded.
-   */
-  public async setFrame(index: number, forceUpdate: boolean = false): Promise<void> {
+  public async setFrame(index: number, forceUpdate: boolean = false): Promise<FrameLoadResult | null> {
     const dataset = this.params?.dataset;
-    // Ignore same or bad frame indices
-    if (!dataset || (!forceUpdate && this.currentFrame === index) || !dataset.isValidFrameIndex(index)) {
-      return;
+    if (!dataset || !dataset.isValidFrameIndex(index)) {
+      // Out of bounds
+      return null;
```

Comment:
why not have this return a FrameLoadResult also?

Reply:
I wanted to distinguish between three states, failed loading (no dataset or invalid frame), partial success (FrameLoadResult but some flags indicate errors) or full success. I guess I could make FrameLoadResult have representations for all three, but the first (failed loading) means that we didn't even try because the parameters were incorrect/invalid.